### PR TITLE
Oracle kvm

### PR DESCRIPTION
**Pull request by eworm-de**

### CODE REVIEW

File: src/basic/virt.c [hunk 1]
```diff
@@ -234,10 +234,28 @@ int detect_vm(const char **id) {
 
         /* this will set _id to "other" and return 0 for unknown hypervisors */
         r = detect_vm_cpuid(&_id);
-        if (r != 0)
+
+        /* finish when found a known hypervisors other than kvm */
+        if (r != 0 && strcmp(_id, "kvm") != 0)
```
Comment (zonque) [inline src/basic/virt.c]: Please use `streq()` here and below.

File: src/basic/virt.c [hunk 2]
```diff
@@ -234,10 +234,28 @@ int detect_vm(const char **id) {
...
                 goto finish;
+        else
+                _id_cpuid = _id;
 
         r = detect_vm_dmi(&_id);
+
+        /* kvm with and without Virtualbox */
+        if (_id_cpuid != NULL && strcmp(_id_cpuid, "kvm") == 0) {
```
Comment (zonque) [inline src/basic/virt.c]: `streq_ptr()` does the `NULL` pointer check for you

**Comment by eworm-de** [comment]
Updated to use `streq()` and `streq_ptr()`.

**Comment by eworm-de** [comment]
Updated to...
- return `oracle` for legacy virtualization and kvm hypervisor
- check all `detect_vm_*()` against `r > 0`
- removed `Signed-Off-by`

**Comment by zonque** [comment]
Looks good to me. @poettering ?

**Comment by eworm-de** [comment]
Ok, next try. I think we got all the error pathes now.

**Comment by poettering** [comment]
Looks good to merge if semaphore agress too.

**Pull request merged**